CSE/Logic/DataManagement/TextProcessing.cs

```diff
@@ -4,11 +4,28 @@
 using System.Text.RegularExpressions;
 using System.Globalization;
 using Logic.Metadata;
+using Logic.Database;
 
 namespace Logic.ImageAnalysis
 {
-    class TextProcessing
+    public class TextProcessing
     {
+		public IEnumerable<string> _productDB;
+
+		public IEnumerable<string> ProductDB //Lazy initialization
+		{
+			get
```

Comment:
Shouldn't this get with if be in Logic/Database/Reader.cs file like a separate method